distribution.md

```diff
@@ -43,29 +44,223 @@ Each buildpack layer blob MUST contain a single buildpack at the following file
 
-Label: `io.buildpacks.buildpackage.metadata`
+#### Labels
+
+The following labels MUST be set in the buildpack image(through the image config's `Labels` field):
+
+| Label                                | Description                                                                                       |
+| ------------------------------------ | ------------------------------------------------------------------------------------------------- |
+| `io.buildpacks.buildpack.metadata` | A JSON object representing information about the packaged entrypoint buildpack                    |
+| `io.buildpacks.buildpack.layers`   | A JSON object representing information about all packaged buildpacks and their associated layers. |
+
+##### `io.buildpacks.buildpack.metadata` (JSON)
+
+```json
+{
+  "id": "<buildpack ID (required)>",
+  "name": "<buildpack name (optional)>",
+  "version": "<buildpack version (required)>",
+  "homepage": "<buildpack home page (optional)>",
+}
+```
+
+##### `io.buildpacks.buildpack.layers` (JSON)
+
 ```json
 {
-  "id": "<entrypoint buildpack ID>",
-  "version": "<entrypoint buildpack version>",
-  "stacks": [
-    {
-      "id": "<stack ID>",
-      "mixins": ["<mixin name>"]
+  # schema of a meta-buildpack
+  "<buildpack ID (required)>": {
+    "<buildpack version (required)>": {
+      "api": "<buildpack API (required)>",
+      "order": [
+        {
+          "group": [
+            {
+              "id": "<inner buildpack ID (required)>",
+              "version": "<inner buildpack version (required)>"
+            }
+          ]
+        }
+      ],
+      "layerDiffID": "<diff ID of buildpack layer (required)>",
+      "homepage": "<buildpack homepage (optional)>",
+      "name": "<buildpack name (optional)>"
+    }
+  },
+  # schema of a regular buildpack
+  "<buildpack ID (required)>": {
+    "<buildpack version (required)>": {
+      "api": "<buildpack API (required)>",
+      "layerDiffID": "<diff ID of buildpack layer (required)>",
+      "homepage": "<buildpack homepage (optional)>",
+      "name": "<buildpack name (optional)>"
     }
-  ]
+  }
 }
 ```
 
+For each buildpack layer, the buildpack ID and the buildpack version MUST be provided in `io.buildpacks.buildpack.layers`.
+
 The buildpack ID and version MUST match a buildpack provided by a layer blob.
 
-For a buildpackage to be valid, each `buildpack.toml` describing a buildpack implementation MUST have all listed stacks.
+For a buildpackage to be valid, each `buildpack.toml` describing a buildpack implementation MUST have all listed targets.
+
+### Builder
+
+The following specifies the artifact format for a buildpacks builder.
+
+A builder is an OCI Image that provides a distributable build environment.
+
+A platform supporting builders SHOULD allow users to configure the build environment for a provided builder.
+
+A platform supporting builder creation MUST be able to create a valid builder from a valid builder.toml file.
```

Comment:
yea, this is good. can we link to the section below? Also, not sure how to say but we don't need to require that the file be _named_ `builder.toml`.

Reply:
cool - i mighta got just a little cheeky and dropped in a note about roses,  but i gave a link so anyone who didn't previously read shakespeare could pick up the reference. lmk if that's too informal for the spec, I can tone it back down to corporate beige.